pkg/proxy/iptables/proxier.go

```diff
@@ -806,15 +808,14 @@ func (proxier *Proxier) syncProxyRules() {
 		return
 	}
 
-	// The value of proxier.needFullSync may change before the defer funcs run, so
-	// we need to keep track of whether it was set at the *start* of the sync.
-	tryPartialSync := !proxier.needFullSync
-
 	// Keep track of how long syncs take.
 	start := time.Now()
+
+	doFullSync := proxier.needFullSync || (start.Sub(proxier.lastFullSync) > proxyutil.FullSyncPeriod)
```

Comment:
```suggestion
	doFullSync := proxier.needFullSync || (time.Since(proxier.lastFullSync) > proxyutil.FullSyncPeriod)
```

it is easier to read  https://go.dev/play/p/a6IUYQjWD0d

Reply:
oh, hold on, it was already discussed https://github.com/kubernetes/kubernetes/pull/129329#discussion_r1905555924, then I leave it to @danwinship suggestion

Reply:
done